src/main/java/org/checkerframework/specimin/MethodPrunerVisitor.java

```diff
@@ -106,4 +118,61 @@ public Visitable visit(ConstructorDeclaration constructorDecl, Void p) {
       return constructorDecl;
     }
   }
+
+  @Override
+  public Visitable visit(FieldDeclaration fieldDecl, Void p) {
+    String classFullName = fieldDecl.resolve().declaringType().getQualifiedName();
+    boolean isFinal = fieldDecl.isFinal();
+    Iterator<VariableDeclarator> iterator = fieldDecl.getVariables().iterator();
+    while (iterator.hasNext()) {
+      VariableDeclarator varDecl = iterator.next();
+      String varFullName = classFullName + "#" + varDecl.getNameAsString();
+
+      if (methodsToEmpty.contains(varFullName)) {
+        varDecl.removeInitializer();
+        if (isFinal) {
+          varDecl.setInitializer(getBasicInitializer(varDecl.getType()));
+        }
+      } else {
+        iterator.remove();
+      }
+    }
+
+    return super.visit(fieldDecl, p);
+  }
+
+  /**
+   * Creates a basic initializer expression for a specified field type. The way the initial value is
+   * chosen is based on the document of the Java Language:
+   * https://docs.oracle.com/javase/specs/jls/se7/html/jls-4.html#jls-4.12.5
+   *
+   * @param fieldType The type for which to generate the basic initializer.
+   * @return An Expression representing the basic initializer for the given field type.
+   */
+  private Expression getBasicInitializer(Type fieldType) {
+    if (fieldType.isPrimitiveType()) {
+      PrimitiveType.Primitive primitiveType = ((PrimitiveType) fieldType).getType();
+      switch (primitiveType) {
+        case BOOLEAN:
+          return new BooleanLiteralExpr(false);
+        case INT:
+          return new IntegerLiteralExpr("0");
+        case LONG:
+          return new LongLiteralExpr("0L");
+        case FLOAT:
+          return new DoubleLiteralExpr("0.0f");
+        case DOUBLE:
+          return new DoubleLiteralExpr("0.0");
+        case BYTE:
+          return new IntegerLiteralExpr("0");
+        case SHORT:
+          return new IntegerLiteralExpr("0");
+          // If none of the above cases are true, then this type is a char
```

Comment:
I'd prefer an explicit case for `CHAR` and a crash if it's something else, which is more future proof. (Maybe a future version of Java will add another primitive type...)